ext/dom/php_dom.c

```diff
@@ -1270,10 +1270,15 @@ xmlNode *dom_get_elements_by_tag_name_ns_raw(xmlNodePtr nodep, char *ns, char *l
 {
 	xmlNodePtr ret = NULL;
 
+	/* Note: The spec says that ns == NULL should change the namespace to the empty string.
+	 *       This is already done at the PHP_FUNCTIONs that create this iterator.
+	 *       The null case here actually means we want to match any namespace instead. */
```

Comment:
So a NULL pointer can only happen via an internal call? 

And the spec says
> If namespace is the empty string, then set it to null. 

Which seems to contradict this comment or am I missing something?

Reply:
> So a NULL pointer can only happen via an internal call?

Yes it can only happen via an internal call.

> Which seems to contradict this comment or am I missing something?

Errr oops, I meant to state they are equivalent.

Reply:
Okay I rewrote the comment to be more clear.